### ISSUE TITLE
Simple Dilbert null pointer exception crash

### ISSUE THREAD

**Issue by billthefarmer** [smarek/Simple-Dilbert#23]
I have a WiFi range extender installed where I usually leave my Nexus 7 tablet charging. I have had Simple Dilbert crash twice now when I go and get the tablet and take it near the WiFi router and start the app. The display times out and says it can't connect to dilbert.com, I select refresh from the menu and get an immediate 'Simple Dilbert has stopped' crash. The first time it happened I couldn't find the traceback, this time I got it. The extender has the same SSID and key as the router, but a different mac address. Other apps I use seem to cope with this. I can't see anything to do with networking in the traceback, so it might be a [red herring](https://en.wikipedia.org/wiki/Red_herring). The app works fine when started again.
```
I/ActivityManager(  502): Start proc com.mareksebera.simpledilbert for activity com.mareksebera.simpledilbert/.core.DilbertFragmentActivity: pid=1081 uid=10099 gids={50099, 3003, 1028, 1015}
E/AndroidRuntime( 1081): FATAL EXCEPTION: main
E/AndroidRuntime( 1081): Process: com.mareksebera.simpledilbert, PID: 1081
E/AndroidRuntime( 1081): java.lang.NullPointerException
E/AndroidRuntime( 1081): 	at com.nostra13.universalimageloader.cache.disc.naming.HashCodeFileNameGenerator.generate(HashCodeFileNameGenerator.java:27)
E/AndroidRuntime( 1081): 	at com.nostra13.universalimageloader.cache.disc.impl.BaseDiscCache.getFile(BaseDiscCache.java:168)
E/AndroidRuntime( 1081): 	at com.nostra13.universalimageloader.cache.disc.impl.BaseDiscCache.remove(BaseDiscCache.java:148)
E/AndroidRuntime( 1081): 	at com.mareksebera.simpledilbert.core.DilbertFragment.refreshAction(DilbertFragment.java:241)
E/AndroidRuntime( 1081): 	at com.mareksebera.simpledilbert.core.DilbertFragment.onOptionsItemSelected(DilbertFragment.java:225)
E/AndroidRuntime( 1081): 	at android.support.v4.app.Fragment.performOptionsItemSelected(Fragment.java:1894)
E/AndroidRuntime( 1081): 	at android.support.v4.app.FragmentManagerImpl.dispatchOptionsItemSelected(FragmentManager.java:2034)
E/AndroidRuntime( 1081): 	at android.support.v4.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:356)
E/AndroidRuntime( 1081): 	at android.support.v7.app.ActionBarActivity.onMenuItemSelected(ActionBarActivity.java:155)
E/AndroidRuntime( 1081): 	at android.support.v7.app.ActionBarActivityDelegate$1.onMenuItemSelected(ActionBarActivityDelegate.java:74)
E/AndroidRuntime( 1081): 	at android.support.v7.app.ActionBarActivityDelegateBase.onMenuItemSelected(ActionBarActivityDelegateBase.java:556)
E/AndroidRuntime( 1081): 	at android.support.v7.internal.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:802)
E/AndroidRuntime( 1081): 	at android.support.v7.internal.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:153)
E/AndroidRuntime( 1081): 	at android.support.v7.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:949)
E/AndroidRuntime( 1081): 	at android.support.v7.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:939)
E/AndroidRuntime( 1081): 	at android.support.v7.internal.view.menu.MenuPopupHelper.onItemClick(MenuPopupHelper.java:187)
E/AndroidRuntime( 1081): 	at android.widget.AdapterView.performItemClick(AdapterView.java:298)
E/AndroidRuntime( 1081): 	at android.widget.AbsListView.performItemClick(AbsListView.java:1113)
E/AndroidRuntime( 1081): 	at android.widget.AbsListView$PerformClick.run(AbsListView.java:2911)
E/AndroidRuntime( 1081): 	at android.widget.AbsListView$3.run(AbsListView.java:3645)
E/AndroidRuntime( 1081): 	at android.os.Handler.handleCallback(Handler.java:733)
E/AndroidRuntime( 1081): 	at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime( 1081): 	at android.os.Looper.loop(Looper.java:136)
E/AndroidRuntime( 1081): 	at android.app.ActivityThread.main(ActivityThread.java:5146)
E/AndroidRuntime( 1081): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime( 1081): 	at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime( 1081): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:732)
E/AndroidRuntime( 1081): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:566)
E/AndroidRuntime( 1081): 	at dalvik.system.NativeStart.main(Native Method)
I/ActivityManager(  502): Process com.mareksebera.simpledilbert (pid 1081) has died.
```

**Comment by smarek** [comment]
Hi @billthefarmer , thanks for report, it was simple to reproduce the issue and is already fixed with commit 6535c700b60163e554db67714fe3ebaa746671ae
I've also updated whole application (libraries, etc..), how do you use the app, compile yourself or install from F-Droid repository?

**Comment by billthefarmer** [comment]
Hi, I get it from f-droid, I don't compile other people's apps as f-droid do such a good job.
Regards
Bill
On 7 Oct 2015 13:40, Marek Sebera <notifications@github.com> wrote:Hi @billthefarmer , thanks for report, it was simple to reproduce the issue and is already fixed with commit 6535c70
I've also updated whole application (libraries, etc..), how do you use the app, compile yourself or install from F-Droid repository?

**Comment by smarek** [comment]
Cool, I've already notified F-Droid team so the update will be out really soon. Thank you!
